### DISCUSSION TITLE
block-level file sync

### DISCUSSION THREAD

**Discussion by phirestalker** [abraunegg/onedrive#1896] [Q&A] [answered]
I read that OneDrive now supports block-level updating of all files (not just documents).
I have several large files (many GBs). I would like to know if this client is capable of uploading only the changed portions of the file.

**Comment by abraunegg** [comment, marked as answer]
@phirestalker 
Block level sync is not supported by the [OneDrive API](https://docs.microsoft.com/en-us/onedrive/developer/rest-api/getting-started/?view=odsp-graph-online) .

It needs to be supported by:
* The underlying file system
  * Microsoft has the advantage here regarding the development of their NTFS & how Windows integrates with NTFS + how OneDrive is built into Windows
* The OneDrive API

Until the OneDrive API implements block-level API access and there is Linux support for providing block level changes to the API - support of block-level sync will be impossible.
